### PR TITLE
Add support for Gradle configuration cache

### PR DESCRIPTION
**Pull request by davidburstrom**
Fixes https://github.com/szpak/gradle-pitest-plugin/issues/249

### CODE REVIEW

File: src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/Junit5FunctionalSpec.groovy [hunk 1]
```diff
@@ -58,4 +58,21 @@ class Junit5FunctionalSpec extends AbstractPitestFunctionalSpec {
...
+            ExecutionResult result = runTasks('pitest', '--configuration-cache')
+            ExecutionResult result2 = runTasks('pitest', '--configuration-cache')
+        then:
+            !result.failure
+            result.wasExecuted('pitest')
+            !result2.failure
+            result2.wasExecuted('pitest')
+        and:
+            result.standardOutput.contains('Configuration cache entry stored')
+            result2.standardOutput.contains('Reusing configuration cache.')
```
Comment (szpak) [inline src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/Junit5FunctionalSpec.groovy:75]: I'm not sure why the previous version of production code throws:
> ClassCastException: class java.lang.Object cannot be cast to class info.solidsoft.gradle.pitest.PitestPlugin  java.lang.Object is in module java.base of loader 'bootstrap'; info.solidsoft.gradle.pitest.PitestPlugin is in  nnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @1d6d2b74)

but after refactoring the test seems to work :-/.

File: src/main/groovy/info/solidsoft/gradle/pitest/PitestPlugin.groovy [hunk 2]
```diff
@@ -175,8 +175,11 @@ class PitestPlugin implements Plugin<Project> {
         task.additionalClasspath.setFrom({
             List<FileCollection> testRuntimeClasspath = (extension.testSourceSets.get() as Set<SourceSet>)*.runtimeClasspath
             FileCollection combinedTaskClasspath = project.objects.fileCollection().from(testRuntimeClasspath)
+            List<String> listOfStrings = extension.
+                    fileExtensionsToFilter.
+                    getOrElse([])
```
Comment (szpak) [inline src/main/groovy/info/solidsoft/gradle/pitest/PitestPlugin.groovy:180]: Good catch, it doesn't need to be calculated multiple times. Nevertheless `listOfStrings` seems to be a default name generated by IDE. `fileExtensionsToFiler`?

**Review by szpak — CHANGES_REQUESTED** [review]
Thanks for your PR! Just small nitpicking.

**Comment by davidburstrom** [comment]
I suspect the CI failure is not cause by my change: https://ci.appveyor.com/project/szpak/gradle-pitest-plugin/builds/40730790/job/d22xjvnb2w0omapm#L606

File: src/main/groovy/info/solidsoft/gradle/pitest/PitestPlugin.groovy (hunk 2 above)
Comment (davidburstrom) [reply src/main/groovy/info/solidsoft/gradle/pitest/PitestPlugin.groovy:180]: Yes, very much an oversight on my side.

File: src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/Junit5FunctionalSpec.groovy (hunk 1 above)
Comment (davidburstrom) [reply src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/Junit5FunctionalSpec.groovy:75]: Well, I'm no expert on the inner workings of the configuration cache, but I reason that in the prior version, `combinedTaskClasspath.filter` is fed with a lambda that refers to the outer `extension` field, which inherently forces the lambda to have a reference to the plugin object, and which subsequently fails deserializing from the cache. After the refactoring, the `List<String>` value is available standalone. The fact that it also is a slight performance improvement is just icing on the cake 🍰

**Comment by davidburstrom** [comment]
Let's see how the corrections fly.

**Comment by szpak** [comment]
> I suspect the CI failure is not cause by my change: https://ci.appveyor.com/project/szpak/gradle-pitest-plugin/builds/40730790/job/d22xjvnb2w0omapm#L606

No, it is a known issue: https://github.com/hcoles/pitest-plugins/pull/4 . I will fix it together with #290 (also to have the releasing on CI up again(. Once done, I will merge your PR. Thanks for your contribution!

**Review by szpak — APPROVED** [review]

**Pull request merged**